src/Keep.lua

```diff
@@ -758,47 +831,21 @@ function Keep:Release()
 		self._last_save = os.clock() -- reset the auto save timer
 
 		if dataKeyInfo then
-			self._keyInfo =
-				{ -- have to map the tuple to a table for type checking (even though tuples are arrays in lua)
-					CreatedTime = dataKeyInfo.CreatedTime,
-					UpdatedTime = dataKeyInfo.UpdatedTime,
-					Version = dataKeyInfo.Version,
-				}
+			self._keyInfo = { -- have to map the tuple to a table for type checking (even though tuples are arrays in lua)
+				CreatedTime = dataKeyInfo.CreatedTime,
+				UpdatedTime = dataKeyInfo.UpdatedTime,
+				Version = dataKeyInfo.Version,
+			}
 		end
 
-		resolve(dataKeyInfo or {})
+		resolve(self)
 	end)
+
 	self.Saving:Fire(updater)
 
 	self._last_save = os.clock()
 
-	updater
-		:andThen(function()
-			self.OnGlobalUpdate:Destroy()
-
-			self._keep_store._cachedKeepPromises[self:Identify()] = nil
-			self._released = true
-
-			releaseCache[self:Identify()] = nil
-		end)
-		:catch(function(err)
-			local keepStore = self._keep_store
-
-			keepStore._processError("Failed to release: " .. err, 2)
-
-			error(err) -- dont want to silence the error
-		end)
-		:finally(function()
-			Keep._activeSaveJobs -= 1
-		end)
-
-	if not self._released then
-		self.Releasing:Fire(updater) -- unlocked, but not removed internally
-	end
-
-	releaseCache[self:Identify()] = updater
-
-	return updater
+	return self:_release(updater)
```

Comment:
Maybe I am missing something, but we lose the functionality of firing releasing here? I probably am saying this and yo have already fixed it in another commit